portal/server/app/route.ts

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { getDomain, getSubdomainAndPath } from '@lib/domain_parsing'
+import { redirectToAggregatorUrlResponse, redirectToPortalURLResponse } from '@lib/redirects'
+import { getBlobIdLink, getObjectIdLink } from '@lib/links'
+import { resolveAndFetchPage } from '@lib/page_fetching'
+
+export async function GET(req: Request) {
+    const originalUrl = req.headers.get('x-original-url')
+    if (!originalUrl) {
+        throw new Error('No original url found in request headers')
+    }
```

Comment:
Question: why do we need this? do we expect proxies changing the request?

Reply:
`x-original-url` is a custom header that contains the original url that was requested by the browser. 
I had to use it because there was no other way accessing the subdomains.